source/expectations/data_values/ut_data_value_refcursor.tpb

```diff
@@ -138,17 +138,17 @@ create or replace type body ut_data_value_refcursor as
     l_xpath := coalesce(self.exclude_xpath, l_other.exclude_xpath);
     if a_other is of (ut_data_value_refcursor) then
       l_other  := treat(a_other as ut_data_value_refcursor);
-      select count(1)
-        into l_result
+      insert into ut_cursor_data_diff ( row_no )
+      select nvl(exp.row_no, act.row_no)
         from (select case when l_xpath is not null then deletexml( ucd.row_data, l_xpath ) else ucd.row_data end as row_data,
                      ucd.row_no
                 from ut_cursor_data ucd where ucd.cursor_data_guid = self.data_value) exp
         full outer join (select case when l_xpath is not null then deletexml( ucd.row_data, l_xpath ) else ucd.row_data end as row_data,
                                 ucd.row_no
                            from ut_cursor_data ucd where ucd.cursor_data_guid = l_other.data_value) act
          on (exp.row_no = act.row_no)
-       where nvl(dbms_lob.compare(xmlserialize( content exp.row_data no indent), xmlserialize( content act.row_data no indent)),1) != 0
-         and rownum <= 1;
+       where nvl(dbms_lob.compare(xmlserialize( content exp.row_data no indent), xmlserialize( content act.row_data no indent)),1) != 0;
```

Comment:
I would add limit here (in the insert statement), so that we only process first 50 - 100 differences.
Why 50 - 100?
10 seems not enough, but anything more than 50 - 100 doesn't bring much value anyway.
If you have so may differences, you're in serious trouble anyway.

If first 100 rows doesn't help you identify what the problem is - your test is not distilled properly and you're testing all in one go probably.

Reply:
Second thought on this idea.
Compare all rows.
Remember toe count of:
- rows in A
- rows in B
- differences
Report first `x` differences (having `x` 50 .. 100)
Report also number of rows in `A`, number of rows in `B` and total number of differences.

That will give much more information about failed assertions,

Reply:
Had something similar in mind myself - would definitely help to output exact number of mismatches because we got them anyways...